### ISSUE TITLE
FSDP "zero-2" checkpoints don't save

### ISSUE THREAD

**Issue by winglian** [axolotl-ai-cloud/axolotl#1776]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

model checkpoint weights to exist in the output_dir under the checkpoint-### dir

### Current behaviour

doing 

```yaml
fsdp:
  - full_shard
  - auto_wrap
fsdp_config:
  fsdp_limit_all_gathers: true
  fsdp_offload_params: false
  fsdp_sync_module_states: true
  fsdp_use_orig_params: true
  fsdp_cpu_ram_efficient_loading: false
  fsdp_transformer_layer_cls_to_wrap: MistralDecoderLayer
  fsdp_state_dict_type: SHARDED_STATE_DICT
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
```
fails when trying to save the checkpoint trainer json file as the checkpoint dir doesn't exist. attempting to extend the trainer class to ensure the dir exists only results in the training_args.json file in the dir.

### Steps to reproduce

see above

### Which Operating Systems are you using?

- [ ] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
![image](https://github.com/user-attachments/assets/5511b2a3-4747-4a6c-8009-c6cd077bdc5a)

I used your fsdp config and was able to successfully save. Should this issue be closed?

I also tested the inference from root output path, and it works.
